includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -686,6 +683,9 @@ public function process_payment_for_order( $cart, $payment_information ) {
 		$order->update_meta_data( '_intent_id', $intent_id );
 		$order->update_meta_data( '_charge_id', $charge_id );
 		$order->update_meta_data( '_intention_status', $status );
+		if ( ! empty( $currency ) ) {
```

Comment:
With reference to my comment above, we don't need this check and can go ahead with the call itself here.